### PR TITLE
[NuGet Symbol Server] Add download symbols support to Package details page

### PR DESCRIPTION
**Pull request by shishirx34**
This PR adds support to download the snupkg from the package details page.

Addresses #5978  and #5970, also added tests.

The idea is to allow downloading a snupkg if the latest uploaded snupkg is "Available". If there is a newer uploaded snupkg that failed validations or still validating, no download allowed. This is taken care of in the download API and in the package details page view.

image

### CODE REVIEW

File: src/NuGetGallery/Strings.resx [hunk 1]
```diff
@@ -949,4 +949,7 @@ If you would like to update the linked Microsoft account you can do so from the
     <value>The previous package version '{0}' is author signed but the uploaded package is unsigned. To avoid this warning, sign the package before uploading.</value>
     <comment>{0} is the previous package's normalized version.</comment>
   </data>
+  <data name="SymbolsPackage_PackageNotAvailable" xml:space="preserve">
+    <value>No availabe symbol package found for ID {0} and version {1}.</value>
```
Comment (skofman1) [inline src/NuGetGallery/Strings.resx]: please verify text with PM

Comment (shishirx34) [reply src/NuGetGallery/Strings.resx]: @diverdan92 - verify.

**Comment by skofman1** [comment]
Probably should get a different icon. Looks great otherwise!

File: src/NuGetGallery/Strings.resx (hunk 1 above)
Comment (diverdan92) [reply src/NuGetGallery/Strings.resx]: Where is this displayed? Otherwise, verified.

Comment (shishirx34) [reply src/NuGetGallery/Strings.resx]: This is displayed when you try to download via rest endpoint and its not available, or the latest uploaded symbol package is validating/failed validation/deleted.

**Pull request merged**